executor/aggfuncs/func_json_objectagg.go

```diff
@@ -97,12 +104,45 @@ func (e *jsonObjectAgg) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup
 		realVal := value.Clone().GetValue()
 		switch x := realVal.(type) {
 		case nil, bool, int64, uint64, float64, string, json.BinaryJSON, *types.MyDecimal, []uint8, types.Time, types.Duration:
+			if _, ok := p.entries[keyString]; !ok {
+				memDelta += int64(len(keyString))
+				memDelta += getValMemDelta(realVal)
+			}
 			p.entries[keyString] = realVal
 		default:
 			return 0, json.ErrUnsupportedSecondArgumentType.GenWithStackByArgs(x)
 		}
 	}
-	return 0, nil
+	return memDelta, nil
+}
+
+func getValMemDelta(val interface{}) (memDelta int64) {
+	memDelta = DefInterfaceSize
+	switch v := val.(type) {
+	case nil:
+		return memDelta
+	case bool:
+		memDelta += DefBoolSize
+	case int64:
+		memDelta += DefInt64Size
+	case uint64:
+		memDelta += DefUint64Size
+	case float64:
+		memDelta += DefFloat64Size
+	case string:
+		memDelta += int64(len(v))
+	case json.BinaryJSON:
+		memDelta += int64(unsafe.Sizeof(v))
+	case *types.MyDecimal:
+		memDelta += int64(unsafe.Sizeof(v))
```

Comment:
Should we using `memDelta += DefMyDecimalSize` here?

Reply:
@b41sh Do you think we should define all of these type like `DefUint32Size` in the `aggfuncs.go`, maybe only we need to use these const variables.